web/src/transmit.ts

```diff
@@ -65,13 +91,14 @@ export function send_message(request, on_success, error) {
                 }
 
                 const response = channel.xhr_error_message("Error sending message", xhr);
-                error(response, xhr.responseJSON?.code);
+                const parsed = z.object({code: z.string()}).safeParse(xhr.responseJSON);
+                error(response, parsed.success ? parsed.data.code : "");
             },
         });
     });
 }
 
-export function reply_message(opts) {
+export function reply_message(opts: {message: Message; content: string}): void {
```

Comment:
For a potential follow-up, I feel like this `opts` calling convention is a bit janky?

Reply:
Sure, I'll add a commit on the end of this PR to change that